nvtabular/dask/categorify.py

```diff
@@ -57,48 +59,145 @@ def get_categories(self, col, path, cache="disk"):
-        for j, split in enumerate(gb.partition_by_hash([col], split_out[col], keep_index=False)):
+    for i, cat_col in enumerate(cat_cols):
+
+        # Compile aggregation dictionary and add "squared-sum"
+        # column(s) (necessary when `cont_cols` is non-empty)
+        df_gb = gdf[[cat_col] + cont_cols].copy(deep=False)
+        agg_dict = {}
+        agg_dict[cat_col] = ["count"]
+        for col in cont_cols:
+            agg_dict[col] = ["sum"]
+            if sum_sq:
+                name = _make_name(col, "pow2")
+                df_gb[name] = df_gb[col].pow(2)
+                agg_dict[name] = ["sum"]
+
+        # Perform groupby and flatten column index
+        # (flattening provides better cudf support)
+        gb = df_gb.groupby(cat_col, dropna=False).agg(agg_dict)
+        gb.columns = [
+            _make_name(*name) if name[0] == cat_col else _make_name(*((cat_col,) + name))
+            for name in gb.columns.to_flat_index()
+        ]
+        gb.reset_index(inplace=True, drop=False)
+        del df_gb
+
+        # Split the result by the hash value of the categorical column
+        for j, split in enumerate(
+            gb.partition_by_hash([cat_col], split_out[cat_col], keep_index=False)
+        ):
             if on_host:
                 output[k] = split.to_pandas()
             else:
                 output[k] = split
             k += 1
-        gdf.drop(columns=[col], inplace=True)
         del gb
-    del gdf
     return output
 
 
-@annotate("cat_level_2", color="green", domain="nvt_python")
-def _cat_level_2(dfs, col, freq_limit, on_host):
+@annotate("mid_level_groupby", color="green", domain="nvt_python")
+def _mid_level_groupby(dfs, col, cont_cols, agg_list, freq_limit, on_host):
     ignore_index = True
     if on_host:
-        # Pandas groupby does not have `dropna` arg
         gb = cudf.from_pandas(_concat(dfs, ignore_index)).groupby(col, dropna=False).sum()
     else:
         gb = _concat(dfs, ignore_index).groupby(col, dropna=False).sum()
     gb.reset_index(drop=False, inplace=True)
+
+    name_count = _make_name(col, "count")
     if freq_limit:
-        gb = gb[gb["count"] >= freq_limit]
+        gb = gb[gb[name_count] >= freq_limit]
+
+    required = [col]
+    if "count" in agg_list:
+        required.append(name_count)
+
+    ddof = 1
+    for cont_col in cont_cols:
+        name_sum = _make_name(col, cont_col, "sum")
+        if "sum" in agg_list:
+            required.append(name_sum)
+
+        if "mean" in agg_list:
+            name_mean = _make_name(col, cont_col, "mean")
+            required.append(name_mean)
+            gb[name_mean] = gb[name_sum] / gb[name_count]
+
+        if "var" in agg_list or "std" in agg_list:
+            n = gb[name_count]
+            x = gb[name_sum]
+            x2 = gb[_make_name(col, cont_col, "pow2", "sum")]
+            result = x2 - x ** 2 / n
+            div = n - ddof
+            div[div < 0] = 0
+            result /= div
```

Comment:
Good point - I was just using the aggregation logic followed in [dask.dataframe](https://github.com/dask/dask/blob/ad5006fc70a256ceb389a3e42b6f303c61c1721d/dask/dataframe/groupby.py#L333).  That does seem a bit dangerous - Perhaps this should be checking that no element is <1 (rather than <0)?  e.g. `div[div < 1] = 1`